vars/newtestpipeline.groovy

```diff
@@ -0,0 +1,127 @@
+                                set -xe
+                                env
+
+                                export SCT_CLUSTER_BACKEND="${params.backend}"
+                                export SCT_REGION_NAME=${aws_region}
+                                export SCT_CONFIG_FILES="${test_config}"
+                                export SCT_COLLECT_LOGS=false
+
+                                if [[ ! -z "${params.scylla_ami_id}" ]] ; then
+                                    export SCT_AMI_ID_DB_SCYLLA="${params.scylla_ami_id}"
+                                elif [[ ! -z "${params.scylla_version}" ]] ; then
+                                    export SCT_SCYLLA_VERSION="${params.scylla_version}"
+                                elif [[ ! -z "${params.scylla_repo}" ]] ; then
+                                    export SCT_SCYLLA_REPO="${params.scylla_repo}"
+                                else
+                                    echo "need to choose one of SCT_AMI_ID_DB_SCYLLA | SCT_SCYLLA_VERSION | SCT_SCYLLA_REPO"
+                                    exit 1
+                                fi
+
+                                export SCT_FAILURE_POST_BEHAVIOR="${pipelineParams.params.get('post_behaviour', '')}"
+                                export SCT_INSTANCE_PROVISION="${pipelineParams.params.get('provision_type', '')}"
+                                export SCT_AMI_ID_DB_SCYLLA_DESC=\$(echo \$GIT_BRANCH | sed -E 's+(origin/|origin/branch-)++')
+                                export SCT_AMI_ID_DB_SCYLLA_DESC=\$(echo \$SCT_AMI_ID_DB_SCYLLA_DESC | tr ._ - | cut -c1-8 )
+
+                                echo "start test ......."
+                                ./docker/env/hydra.sh run-test ${pipelineParams.test_name} --backend ${params.backend}  --logdir /sct
+                                echo "end test ....."
+                                """
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        post {
+                always {
+                    script {
+                        wrap([$class: 'BuildUser']) {
+                            dir('scylla-cluster-tests') {
+                                def aws_region = groovy.json.JsonOutput.toJson(params.aws_region)
+                                def test_config = groovy.json.JsonOutput.toJson(pipelineParams.test_config)
+
+                                sh """
+                                #!/bin/bash
+
+                                set -xe
+                                env
+
+                                export SCT_CLUSTER_BACKEND="${params.backend}"
+                                export SCT_REGION_NAME=${aws_region}
+                                export SCT_CONFIG_FILES="${test_config}"
+
+                                echo "start collect logs ..."
+                                ./docker/env/hydra.sh collect-logs --logdir /sct
```

Comment:
please add a step for cleaning the resources after the log are collected
I guess it should be a new step, or something, it should happen even if logs are failed to be collected

Reply:
added